packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts

```diff
@@ -0,0 +1,108 @@
+import { builtinModules } from "module";
+import { isAbsolute, relative } from "path";
+
+import type { Plugin } from "esbuild";
+
+import { RemixConfig } from "../../config";
+
+/**
+ * A plugin responsible for resolving bare module ids based on server target.
+ * This includes externalizing for node based plaforms, bundling for single file
+ * environments such as cloudflare, and resolving node built-ins to denoland
+ * URL's when possible.
+ */
+export function serverBareModulesPlugin(
+  remixConfig: RemixConfig,
+  dependencies: Record<string, string>,
+  onWarning?: (warning: string, key: string) => void
+): Plugin {
+  return {
+    name: "bare-modules",
+    setup(build) {
+      build.onResolve({ filter: /.*/ }, ({ importer, path }) => {
+        // If it's not a bare module ID, bundle it.
+        if (!isBareModuleId(path)) {
+          return undefined;
+        }
+
+        // To prevent `import xxx from "remix"` from ending up in the bundle
+        // we "bundle" remix but the other modules where the code lives.
+        if (path === "remix") {
+          return undefined;
+        }
+
+        // These are our virutal modules, always bundle the because there is no
+        // "real" file on disk to externalize.
+        if (
+          path === "@remix-run/server-entry" ||
```

Comment:
Call it `@remix-run/server-build`

Maybe `import { serverBuildVirtualModuleId } from "somewhere"`